File: pset6/credit/credit.py
from cs50 import get_string
def main():
    check=False
    sum=0
    while check==False:
        try:
            a=get_string("Please enter a number: ")
            i=int(a)
            check=True
        except ValueError:
            check=False
    if len(a)!=13 and len(a)!=15 and len(a)!=16:
        print("INVALID\n")
        return
    for x in range(0,len(a)-1,2):
        position=len(a)-x-2
        y=2*int(a[position])
        #print(f"position {position} gives {y}")
        z=y//10 +y%10
        sum+=z
    for x in range(0,len(a),2):
        position=len(a)-x-1
        y=int(a[position])
        #print(f"position {position} gives {y}")
        sum+=y
    #print(sum)
    if sum%10!=0:
        print("INVALID\n")
        return
    start=int(a[0])*10+int(a[1])

    if start==34 or start==37:
        cardid="AMEX"
    elif start>=40 and start<=49:
        cardid="VISA"
    elif start>=51 and start<=55:
        cardid="MASTERCARD"
    else:
        print("INVALID\n")
        return
    print(cardid)

def exponent(base,power):
    result=1;
    for x in range(power):
        result*=base
    return result
if __name__=="__main__":
    main()

File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>


int main(void) {
    bool check = false;
    string a = NULL;
    string id = NULL;
    do {
      a = get_string("Enter in a number\n");
      for (int i = 0; i < strlen(a); i++) {
        if (!(a[i] > 47 && a[i] < 58))
          check = false;
        else
          check = true;
      }

    } while (check == false);

    if (!(strlen(a) == 13 || strlen(a) == 15 || strlen(a) == 16)) {
      printf("INVALID\n");
      return 0;
    }

    int sum = 0;

    for (int i = 1; i < strlen(a); i += 2) {
      int position = strlen(a) - i - 1;
      int x = a[position] - '0';
      int y = 2 * x;
      int z = y / 10 + y % 10;
      sum += z;
    }

    for (int i = 1; i <= strlen(a); i += 2) {
      int position = strlen(a) - i;
      int x = a[position] - '0';
      sum += x;
    }

    if (sum % 10 != 0) {
      printf("INVALID\n");
      return 0;
    }
    int start = (a[0] - '0') * 10 + (a[1] - '0');
    if (start == 34 || start == 37) {
      id = "AMEX";
    } else if (start >= 40 && start <= 49) {
      id = "VISA";
    } else if (start >= 51 && start <= 55) {
      id = "MASTERCARD";
    } else {
      printf("INVALID\n");
      return 0;
    }
    printf("%s\n", id);

    return 0;
}